src/colmap/scene/image.h

```diff
@@ -125,6 +125,14 @@ class Image {
   void SetPoints2D(const std::vector<Eigen::Vector2d>& points);
   void SetPoints2D(const std::vector<struct Point2D>& points);
 
+  std::vector<point3D_t> Point3DIds(
+    const std::optional<std::vector<point2D_t>>& point2D_idxs = std::nullopt
```

Comment:
Otherwise, it's easy to create an unnecessary copy of the input vector.
```suggestion
    const std::vector<point2D_t>* point2D_idxs = nullptr
```